AndroidAnnotations/androidannotations-rest-spring/rest-spring/src/main/java/org/androidannotations/rest/spring/helper/RestSpringValidatorHelper.java

```diff
@@ -601,17 +600,12 @@ public void hasAnnotatedAllParameters(ExecutableElement element, ElementValidati
 		}
 
 		Set<String> urlVariableNames = restAnnotationHelper.extractUrlVariableNames(element);
-		if (urlVariableNames.size() != numberOfNotAnnotatedElementParameter(element) + numberOfPathAnnotatedParameter(element) + numberOfRequiresCookieInUrl(element)) {
-			validation.addError(element, "%s parameters must add annotations or define as @Path placeholders");
+		if (urlVariableNames.size() != numberOfPathAnnotatedParameter(element) + numberOfRequiresCookieInUrl(element)) {
+			validation.addError(element, "%s parameters must add annotations");
```

Comment:
Or there are missing url variables corresponding to the `@RequiresCookieInUrl` annotation.